integration-tests/src/test/java/com/example/IntegrationTest.java

```diff
@@ -285,6 +285,30 @@
     assertThat(values.get("1").get()).isEqualTo("one");
   }
 
+  @Test public void nestedComponent() {
+    NestedComponent.TheComponent component =
+        backend.create(NestedComponent.TheComponent.class);
+    assertThat(component.string()).isEqualTo("foo");
+  }
+
+  @Test public void nestedComponentBuilder() {
+    NestedComponent.TheComponent component =
+        backend.builder(NestedComponent.TheComponent.Builder.class).build();
+    assertThat(component.string()).isEqualTo("foo");
+  }
+
+  @Test public void deeplyNestedComponent() {
```

Comment:
Do we need both of these? Let's just keep the deep one.

Reply:
K, Will revise tonight